lib/db/set.go

```diff
@@ -157,6 +157,8 @@ func (s *FileSet) Update(device protocol.DeviceID, fs []protocol.FileInfo) {
 		}
 		s.blockmap.Discard(discards)
 		s.blockmap.Update(updates)
+		s.db.removeSequences([]byte(s.folder), discards)
```

Comment:
This is neat. I'd been thinking about something similar before but didn't know where to hook in the update code. 👍

Reply:
Honestly my main thought throughout doing this was: Where is the catch? This is too simple to not having been done before.

Reply:
Sorry, I am curious now, what's the complicated bit here? Finding where to do removals?

Reply:
I always imagined doing it in the something-update function in the middle of the database layer, where it would then need to load the previous version to find the old sequence etc. This is cleaner.